src/metaprogramming.ts

```diff
@@ -88,6 +88,22 @@ type DeepPickUnsafe<T, Keys extends string> = {
     : DeepPickUnsafe<T[Key], TailsOf<Keys, Key>>;
 };
 
+/**
+ * Make properties of an object required.
+ *
+ * type Foo = {
+ *     a?: string
+ *     b?: number
+ *     c?: object
+ * }
+ *
+ * type Bar = RequireKeys<Foo, "a" | "b">
+ * // Property "a" and "b" are now required.
+ */
+export type RequireKeys<T extends object, Keys extends keyof T> = T & {
```

Comment:
Would you like me to make a DeepRequire like we have DeepOmit and DeepPick? Then you can say DeepRequire<Foo, "a.b" | "c">